base/range.jl

```diff
@@ -116,9 +116,9 @@ _range(a::T, step::T, ::Nothing, len::Integer) where {T <: AbstractFloat} =
 _range(a::T, step, ::Nothing, len::Integer) where {T} =
     _rangestyle(OrderStyle(T), ArithmeticStyle(T), a, step, len)
 _rangestyle(::Ordered, ::ArithmeticWraps, a::T, step::S, len::Integer) where {T,S} =
-    StepRange{typeof(a+step),S}(a, step, a+step*(len-1))
+    StepRange{typeof(a+false*step),S}(a, step, a+step*(len-1))
```

Comment:
Wouldn't it be clearer to express this as `zero(step)`?

Reply:
Yes, I will change it.